src/unix/tty.c

```diff
@@ -129,8 +141,7 @@ int uv_tty_init(uv_loop_t* loop, uv_tty_t* tty, int fd, int readable) {
     if (newfd != -1)
       uv__close(newfd);
     QUEUE_REMOVE(&tty->handle_queue);
-    if (!(flags & UV_STREAM_BLOCKING))
-      uv__nonblock(fd, 0);
+    fcntl(fd, F_SETFL, saved_flags);
```

Comment:
Ditto about looping on EINTR.  If #885 lands, you could use `uv__nonblock_slow()` instead.